src/Evolution/Systems/Cce/Equations.hpp

```diff
@@ -0,0 +1,1020 @@
+                                                  TagsCategory::Temporary> {
+  using type = tmpl::list<
+      ::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>, 2>>;
+};
+
+// - Tags for Linear operator that acts on the conjugate of H
+template <>
+struct tags_needed_for_integrand_computation_impl<
+    Tags::LinearFactorForConjugate<Tags::H>,
+    TagsCategory::TagsToSwshDifferentiate> {
+  using type = tmpl::list<Tags::Dy<Tags::J>, Tags::J>;
+};
+template <>
+struct tags_needed_for_integrand_computation_impl<
+    Tags::LinearFactorForConjugate<Tags::H>,
+    TagsCategory::IndependentOfBondiIntegration> {
+  using type = tmpl::list<Tags::OneMinusY>;
+};
+template <>
+struct tags_needed_for_integrand_computation_impl<
+    Tags::LinearFactorForConjugate<Tags::H>, TagsCategory::Temporary> {
+  using type = tmpl::list<
+      ::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>, 2>>;
+};
+}  // namespace detail
+
+/// \brief A struct for providing a tmpl::list of required tags for each step in
+/// the CCE computation.
+/// \details Given a tag `BondiVariable` associated with one of the
+/// below Bondi integrand computations, and a `Category` (which may be
+/// `TagsToSwshDifferentiate`, `SwshDerivatives`,
+/// `IndependentOfBondiIntegration`, `Temporary`), this is a type alias to a
+/// tmpl::list of the set of tags of the category accessed in the function which
+/// computes that integrand.
+template <typename BondiVariableIntegrand, typename Category>
+using tags_needed_for_integrand_computation =
+    typename detail::tags_needed_for_integrand_computation_impl<
+        BondiVariableIntegrand, Category>::type;
+
+/// \brief A struct for providing a tmpl::list of integrand tags that need to be
+/// computed before integration can proceed for a given Bondi variable tag.
+template <typename BondiVariable>
+using integrand_terms_to_compute_for_bondi_variable =
+    typename detail::integrand_terms_to_compute_for_bondi_variable_impl<
+        BondiVariable>::type;
+
+/*!
+ * \brief Computes one of the inputs for the integration of one of the
+ * Characteristic hypersurface equations.
+ *
+ * \details The template argument must be one of the integrand-related prefix
+ * tags applied to a Bondi quantity for which that integrand is required. The
+ * relevant prefix tags are `Tags::Integrand`, `Tags::PoleOfIntegrand`,
+ * `Tags::RegularIntegrand`, `Tags::LinearFactor`, and
+ * `Tags::LinearFactorForConjugate`. The Bondi quantity tags that these tags
+ * may wrap are `Tags::Beta`, `Tags::Q`, `Tags::U`, `Tags::W`, and `Tags::H`.
+ *
+ * Additional mathematical details for each of the computations can be found in
+ * the template specializations of this struct. All of the specializations have
+ * a static `apply` function which takes as arguments
+ * `SpinWeighted<ComplexDataVector, N>`s for each of the Bondi arguments.
+ */
+template <typename IntegrandTag>
+struct ComputeBondiIntegrandImpl;
```

Comment:
The convenience impl should instead be a impl function moved in the ComputeBondiIntegrand function.